Formula/snapcraft.rb

```diff
@@ -89,6 +92,12 @@ class Snapcraft < Formula
 
   fails_with gcc: "5" # due to apt on Linux
 
+  # Needs `setuptools<66`
+  resource "setuptools" do
+    url "https://files.pythonhosted.org/packages/a1/29/f2ad3b78b9ebd24afa282eed9add27b47ef52b37291198021154b4b65166/setuptools-65.7.0.tar.gz"
+    sha256 "4d3c92fac8f1118bb77a22181355e29c239cabfe2b9effdaa665c66b711136d7"
+  end
```

Comment:
Didn't work. 😅

Reply:
Looks like `--ignore-installed` prevents pip from removing the newer version.

Reply:
Worked for pip at https://github.com/Homebrew/homebrew-core/pull/134447 though.

Reply:
Then might be specific to `setuptools` or some undefined behavior.

In case of `pip`, I locally tried installing older `pip` in venv and the output version numbers are all over the place (`pip install` shows wheel is for 22.3.1 but then claims to have installed 23.0.1. Afterward, `pip list` still shows 23.0.1 but importing pip shows 22.3.1):
```console
❯ pip list
Package    Version
---------- -------
pip        23.0.1
setuptools 67.6.1

❯ pip install -vvv --no-deps --no-binary :all:  --ignore-installed ./pip-22.3.1/
Using pip 23.0.1 from /private/tmp/test-venv/venv/lib/python3.11/site-packages/pip (python 3.11)
...
  Created wheel for pip: filename=pip-22.3.1-py3-none-any.whl size=2051532 sha256=fdc1a1a2e7a8d5285b5d60bc1a1e6344845e395df4d70049eb69dcf21f9fe7e6
  Stored in directory: /Users/cho-m/Library/Caches/pip/wheels/c0/bd/3d/b1f9df2824bf1974a39b5c47d6b437cb4fd7628bcb5f486efa
Successfully built pip
Installing collected packages: pip

  changing mode of /private/tmp/test-venv/venv/bin/pip to 755
  changing mode of /private/tmp/test-venv/venv/bin/pip3 to 755
  changing mode of /private/tmp/test-venv/venv/bin/pip3.11 to 755
Successfully installed pip-23.0.1
Remote version of pip: 23.1.2
Local version of pip:  23.0.1
Was pip installed by pip? True

❯ pip list
Package    Version
---------- -------
pip        23.0.1
setuptools 67.6.1

❯ python3 -c 'import pip; print(pip.__version__)'
22.3.1
```

Only difference for setuptools is imported version is also unchanged:
```console
❯ pip install -vvv --no-deps --no-binary :all:  --ignore-installed ./setuptools-65.7.0/
Using pip 22.3.1 from /private/tmp/test-venv/venv/lib/python3.11/site-packages/pip (python 3.11)
...
  Created wheel for setuptools: filename=setuptools-65.7.0-py3-none-any.whl size=1234046 sha256=0f45194ed9132662327a88c9484237cfddbf7e852866a9d0479f99cc41cb79b2
  Stored in directory: /Users/cho-m/Library/Caches/pip/wheels/04/97/31/286624957c941070bd12d9bdf8c69f98b7306bd4aadbdf9067
Successfully built setuptools
Installing collected packages: setuptools

Successfully installed setuptools-67.6.1

❯ pip list
Package    Version
---------- -------
pip        23.0.1
setuptools 67.6.1

❯ python3 -c 'import setuptools; print(setuptools.__version__)'
67.6.1
```